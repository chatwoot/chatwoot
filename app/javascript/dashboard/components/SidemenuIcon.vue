<template>
  <woot-button
    size="small"
    variant="clear"
    color-scheme="secondary"
    icon="list"
    class="toggle-sidebar"
    @click="onMenuItemClick"
  />
</template>

<script>
import { BUS_EVENTS } from 'shared/constants/busEvents';

export default {
  methods: {
    onMenuItemClick() {
      bus.$emit(BUS_EVENTS.TOGGLE_SIDEMENU);
    },
  },
};
</script>
<style scoped lang="scss">
.toggle-sidebar {
  margin-left: var(--space-minus-small);
}
</style>
