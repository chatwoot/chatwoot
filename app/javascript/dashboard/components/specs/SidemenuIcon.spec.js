import { shallowMount } from '@vue/test-utils';
import SidemenuIcon from '../SidemenuIcon';

describe('SidemenuIcon', () => {
  test('matches snapshot', () => {
    const wrapper = shallowMount(SidemenuIcon);
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
