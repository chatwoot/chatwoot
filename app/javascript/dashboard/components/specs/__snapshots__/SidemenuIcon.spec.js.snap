// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidemenuIcon matches snapshot 1`] = `
<woot-button
  class="toggle-sidebar"
  color-scheme="secondary"
  icon="list"
  size="small"
  variant="clear"
/>
`;
